apps/event_sourcing_core/src/event_sourcing_core_store_ets.erl

```diff
@@ -117,17 +117,23 @@ event_to_record(Event) ->
         event = Event
     }.
 
--spec save_snapshot(Snapshot) -> ok when
-    Snapshot :: snapshot().
+-spec save_snapshot(Snapshot) -> ok | {error, Reason} when
```

Comment:
On further consideration, since missing a snapshot isn't critical, `{warning, Reason}` would be clearer than `{error, Reason}` and avoid misleading callers into treating it as fatal.